src/java/org/apache/cassandra/db/DeletionTime.java

```diff
@@ -195,42 +195,126 @@ public static Serializer getSerializer(Version version)
             return legacySerializer;
     }
 
-    // Serializer for Usigned Integer ldt
+    /* Serializer for Usigned Integer ldt
+     *
+     * ldt is encoded as a uint in seconds since unix epoch, it can go up o 2106-02-07T06:28:13+00:00 only. 
+     * Since mfda is micros since the Unix Epoch with 7 bytes we can encode 2^56 ~= 2284 years. That leaves 
+     * 1 free byte we can use to store flags for sentinel values with some space saving.
+     *
+     * Currently we flag:
+     *  - A LIVE ldt 0x01
+     *  - A maxed mfda 0x02
+     */
     public static class Serializer implements ISerializer<DeletionTime>
     {
+        private final static int IS_LIVE_DELETION = 0x01;
+        private final static int IS_MAX_MFDA = 0x02;
+        // Long.MAX_VALUE sentinel value equivalent in 7bytes
+        public final static long MAX_MFDA = (1L << (7 * 8)) - 1;
+
         public void serialize(DeletionTime delTime, DataOutputPlus out) throws IOException
         {
-            out.writeInt(delTime.localDeletionTimeUnsignedInteger);
-            out.writeLong(delTime.markedForDeleteAt());
+            if (delTime.equals(LIVE))
```

Comment:
this is very nit, don't bother if you don't want - should it be enough to just check `delTime == LIVE`?

Reply:
Yeah this is my OCD kicking in. If we ever changed DeletionTime in a way we want _content_ comparison, overload `equals`.... I prefer to leave it as it is.

Reply:
then, maybe add `if (o == this) return true` to `DeletionTime.equals`. Checking `instanceof` is much less efficient

Reply:
Ok I'll do `==` then

Reply:
Fixed in latest commit. Feel free to reopen if needed.